caravel/views.py

```diff
@@ -901,130 +941,89 @@ def explore(self, datasource_type, datasource_id, slice_id=None):
         datasource = datasource[0] if datasource else None
 
         if not datasource:
-            flash(__("The datasource seems to have been deleted"), "alert")
+            flash(DATASOURCE_MISSING_ERR, "alert")
             return redirect(error_redirect)
 
-        all_datasource_access = self.can_access(
-            'all_datasource_access', 'all_datasource_access')
-
-        datasource_access = self.can_access(
-            'datasource_access', datasource.perm)
-
-        if not (all_datasource_access or datasource_access):
-            flash(__("You don't seem to have access to this datasource"),
-                  "danger")
+        if not self.datasource_access(datasource):
+            flash(__(get_datasource_access_error_msg(datasource)), "danger")
             return redirect(error_redirect)
 
-        # handle slc / viz obj
-        slice_id = slice_id or request.args.get("slice_id")
-        viz_type = None
-        slc = None
-
-        slice_params = request.args
+        request_args_multi_dict = request.args  # MultiDict
 
+        slice_id = slice_id or request_args_multi_dict.get("slice_id")
+        slc = None
+        # build viz_obj and get it's params
         if slice_id:
-            slc = (
-                db.session.query(models.Slice)
-                .filter_by(id=slice_id)
-                .first()
-            )
+            slc = db.session.query(models.Slice).filter_by(id=slice_id).first()
             try:
-                param_dict = json.loads(slc.params)
-
+                viz_obj = slc.get_viz(
+                    url_params_multidict=request_args_multi_dict)
             except Exception as e:
                 logging.exception(e)
-                param_dict = {}
-
-            # override slice params with anything passed in url
-            # some overwritten slices have been saved with original slice_ids
-            param_dict["slice_id"] = slice_id
-            param_dict["json"] = "false"
-            param_dict.update(request.args.to_dict(flat=False))
-            slice_params = ImmutableMultiDict(param_dict)
+                flash(utils.error_msg_from_exception(e), "danger")
+                return redirect(error_redirect)
+        else:
+            viz_type = request_args_multi_dict.get("viz_type", None)
```

Comment:
Unless multidict has a subtlety when compared to a normal dict, the second argument `None` is implicit as the get method returns None if the key is not found by default